Updated docs on CopyTo(Stream, Int32)

Shailosingh opened the pull request:
## Summary
This method reads a specified number of bytes from the current stream, advances the read position and then writes them to the destination stream, advancing the write position of that as well.

The old documentation did not make it clear that the write or read positions were advanced at all. This can lead to a lot of confusion for programmers when they attempt to use this method.

opbld34 commented:
Docs Build status updates of commit _[a4e9991](https://github.com/Shailosingh/dotnet-api-docs/commits/a4e9991ba94d0a0f61cbf12cedaf364e3f3446d4)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.IO/Stream.xml](https://github.com/Shailosingh/dotnet-api-docs/blob/patch-1/xml/System.IO/Stream.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.io.stream?view=dotnet-uwp-10.0&branch=pr-en-us-7966) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2022%5C4%5C17%5C500c749d-de78-3d4a-00e4-269d599bf9ec%5CPullRequest%5C202204172222344427-7966%5Cworkflow_report.html?sv=2021-04-10&se=2022-05-18T22%3A47%3A04Z&sr=b&sp=r&sig=cZv723FOmQptO%2BOTdStnwe8PKYgwLxAx58rcLbITeW0%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

gewarren reviewed (commented):
If we update one overload, we should update all of the overloads. But I'd like the SMEs to weigh in first.

adamsitnik commented on xml/System.IO/Stream.xml:
```diff
@@ -775,7 +775,7 @@
         <param name="destination">The stream to which the contents of the current stream will be copied.</param>
         <param name="bufferSize">The size of the buffer. This value must be greater than zero. The default size is 81920.</param>
-        <summary>Reads the bytes from the current stream and writes them to another stream, using a specified buffer size.</summary>
+        <summary>Reads the bytes from the current stream, advances the current stream's read position, writes the bytes to the destination stream and advances the destination stream's write position, using a specified buffer size.</summary>
```
```suggestion
        <summary>Reads the bytes from the current stream and writes them to another stream, using a specified buffer size. Both streams positions are advanced by the number of bytes copied.</summary>
```

Shailosingh replied:
Thank you very much! The simplification makes it much easier to understand.

adamsitnik reviewed (commented):
@Shailosingh big thanks for your contribution!

Overall, it looks good to me, but personally I would simplify it. Please take a look at my suggestion and let me know what do you think.

opbld34 commented:
Docs Build status updates of commit _[d3719eb](https://github.com/Shailosingh/dotnet-api-docs/commits/d3719eb4e7063c377425eec29d1bba7f4be0df0e)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.IO/Stream.xml](https://github.com/Shailosingh/dotnet-api-docs/blob/patch-1/xml/System.IO/Stream.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.io.stream?view=dotnet-uwp-10.0&branch=pr-en-us-7966) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2022%5C4%5C20%5C500c749d-de78-3d4a-00e4-269d599bf9ec%5CPullRequest%5C202204202047514491-7966%5Cworkflow_report.html?sv=2021-04-10&se=2022-05-21T21%3A10%3A43Z&sr=b&sp=r&sig=GmR68gTAmRgdSvM7T8YBd84xyKhE9B%2Bnmzg0rsFTaMA%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

Shailosingh commented:
@gewarren Thank you very much for the feedback! I think I updated the overloads and also updated the definition of Stream.CopyTo as well. I'm still pretty new to submitting pull requests to big repositories, so please let me know if something is wrong.

opbld34 commented:
Docs Build status updates of commit _[b11e65f](https://github.com/Shailosingh/dotnet-api-docs/commits/b11e65f1ce4daa4c426590e83f61bbb6cfc030af)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.IO/Stream.xml](https://github.com/Shailosingh/dotnet-api-docs/blob/patch-1/xml/System.IO/Stream.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.io.stream?view=dotnet-uwp-10.0&branch=pr-en-us-7966) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2022%5C4%5C20%5C500c749d-de78-3d4a-00e4-269d599bf9ec%5CPullRequest%5C202204202118128549-7966%5Cworkflow_report.html?sv=2021-04-10&se=2022-05-21T21%3A46%3A09Z&sr=b&sp=r&sig=JWVQj7sCSM9mYalAUfSc45x3%2FWdjecow8M7eED5n398%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

adamsitnik reviewed (commented):
@Shailosingh the synchronous overloads look good to me. Could you please add it to async one too? You can find them by searching for `<Member MemberName="CopyToAsync">`

again, thank you for your contribution!

Shailosingh commented:
@adamsitnik Thank you for the suggestion. I have now adjusted the CopyToAsync overloads and definition 😊.

opbld34 commented:
Docs Build status updates of commit _[5b98982](https://github.com/Shailosingh/dotnet-api-docs/commits/5b98982d2e0eeb93de091c4acdcdff3d8f79a5d5)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.IO/Stream.xml](https://github.com/Shailosingh/dotnet-api-docs/blob/patch-1/xml/System.IO/Stream.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.io.stream?view=dotnet-uwp-10.0&branch=pr-en-us-7966) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2022%5C4%5C23%5C500c749d-de78-3d4a-00e4-269d599bf9ec%5CPullRequest%5C202204230158090901-7966%5Cworkflow_report.html?sv=2021-04-10&se=2022-05-24T02%3A26%3A46Z&sr=b&sp=r&sig=m8J%2FCQ8NaomJI6Y5AHLygQUdg%2FqDnwgYVVNwKtU27mA%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

adamsitnik reviewed (approved):
LGTM, thank you @Shailosingh !
